libs/ledger-live-common/src/families/solana/banner.ts

```diff
@@ -0,0 +1,84 @@
+import { getCurrentSolanaPreloadData } from "./js-preload-data";
+import { stakeActions } from "./logic";
+import { isAccountEmpty } from "../../account";
+import { LEDGER_VALIDATOR_ADDRESS } from "./utils";
+import type { SolanaAccount } from "./types";
+import { ValidatorsAppValidator } from "./validator-app";
+
+export interface AccountBannerState {
+  display: boolean;
+  redelegate: boolean;
+  stakeAccAddr: string;
+  ledgerValidator: ValidatorsAppValidator | undefined;
+}
+
+export function getAccountBannerState(
+  account: SolanaAccount
+): AccountBannerState {
+  // Group current validator
+  const solanaResources = account.solanaResources
+    ? account.solanaResources
+    : { stakes: [] };
+  const delegations = solanaResources?.stakes.map((delegation) => {
+    return delegation;
+  });
+
+  // Get ledger validator data
+  const { validators } = getCurrentSolanaPreloadData(account.currency) ?? {
+    validators: [],
+  };
+  console.log("validators", validators);
+  const ledgerValidator = validators.find(
+    (validator) => validator.voteAccount === LEDGER_VALIDATOR_ADDRESS
+  );
+
+  // If Ledger doesn't provide validator, we don't display banner
+  if (!ledgerValidator) {
+    return {
+      display: false,
+      redelegate: false,
+      stakeAccAddr: "",
+      ledgerValidator,
+    };
+  }
+
+  let redelegate = false;
+  let stakeAccAddr = "";
+  let display = false;
+
+  // Find user current worst validator (default validator is ledger)
+  let worstValidator = ledgerValidator;
+  for (let i = 0; i < delegations.length; i++) {
+    const validatorAdress = delegations[i]?.delegation?.voteAccAddr;
+    const validator = validators.find(
+      (validator) => validator.voteAccount === validatorAdress
+    );
+    const actions = stakeActions(delegations[i]);
+    const isValidRedelegation =
+      validator &&
+      validatorAdress !== ledgerValidator.voteAccount &&
+      worstValidator.commission <= validator.commission &&
+      actions.includes("deactivate");
+    if (isValidRedelegation) {
+      stakeAccAddr = delegations[i].stakeAccAddr;
```

Comment:
```suggestion
  for (const delegation of delegations) {
    const validatorAdress = delegation.delegation?.voteAccAddr;
    const validator = validators.find(
      (validator) => validator.voteAccount === validatorAdress
    );
    const actions = stakeActions(delegation);
    const isValidRedelegation =
      validator &&
      validatorAdress !== ledgerValidator.voteAccount &&
      worstValidator.commission <= validator.commission &&
      actions.includes("deactivate");
    if (isValidRedelegation) {
      stakeAccAddr = delegation.stakeAccAddr;
```

Reply:
I think `for of` reads a bit better, and you don't need the null check of line 52